### PR TITLE
feat: add minimal next gen 'activate' command

### PR DESCRIPTION
**Pull request by zmitchell**
## Proposed Changes

This PR implements a very minimal, feature-flagged `flox activate` that simply calls `nix shell` under the hood, using the flake in the `.flox` directory.

## Current Behavior

As part of #202 `flox activate` called with the above feature flag simply calls `nix shell`. This simply moves the implementation to a different place.

## Checks

- [x] All tests pass.
- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] I have accepted the flox [Contributor License Agreement](../blob/main/.github/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](../blob/main/.github/CONTRIBUTORS.csv) file by way of this pull request or one done previously.

## Release Notes

N/A

### CODE REVIEW

**Review by tomberek — APPROVED** [review]

**Comment by zmitchell** [comment]
Discussed with @ysndr and @mkenigs earlier today, this PR will be rebased and merged after #215 is merged since it will be easier to rebase this PR instead of that one.

File: tests/environment-activate.bats
Comment (mkenigs) [inline tests/environment-activate.bats:181]: If it's not painful, could you add one test that checks that PATH is modified?

Comment (zmitchell) [reply tests/environment-activate.bats:181]: Sure thing

File: crates/flox-rust-sdk/src/models/environment.rs
Comment (mkenigs) [inline crates/flox-rust-sdk/src/models/environment.rs:282]: As this is starting a subshell I think I would expect that it's in the flox crate? I never felt fully resolved about https://github.com/flox/flox/commit/88f147c1783e0b0eefcc7d7a20ab4691a2c68c0c#r124910472

Comment (zmitchell) [reply crates/flox-rust-sdk/src/models/environment.rs:282]: How do you expect that to look? The `activate` method is on the `Environment` trait, which lives in `flox-rust-sdk`. I think it makes less sense to have the interface for an environment split across multiple crates. Given that we've already decided that this is a temporary solution, I'd prefer to just stick with this until we have the actual `activate` implementation figured out.

Comment (zmitchell) [reply crates/flox-rust-sdk/src/models/environment.rs:282]: If you're going to have a (self-imposed) boundary inside the flox application, I think it makes more sense for flox to be the coordinator and the interface, calling high level methods on the lower level components. It feels arbitrary to me to decide that nothing outside of flox can create a shell.

I'm not trying to pick another fight here, I'm just pointing out that where we decide to put this boundary has implications for simplicity of the interfaces and how natural they feel.

**Comment by stahnma** [comment]
Possibly blocked on flox #215

**Review by mkenigs — APPROVED** [review]

**Pull request merged**
